pkg/client/results/processing.go

```diff
@@ -146,48 +148,120 @@ func PostProcessPlugin(p plugin.Interface, dir string) (Item, error) {
 	return i, err
 }
 
+// processNodesWithProcessor is called to invoke processDir on each node-specific directory contained
+// underneath the given dir. The directory is assumed to be either the results directory or errors directory
+// which should have the nodes as subdirectories. It returns an item for each node processed and an error
+// only if it couldn't open the original directory. Any errors while processing a specific node are logged
+// but not returned.
+func processNodesWithProcessor(p plugin.Interface, baseDir, dir string, processor postProcessor, selector fileSelector) ([]Item, error) {
+	pdir := path.Join(baseDir, PluginsDir, p.GetName())
+
+	nodeDirs, err := ioutil.ReadDir(dir)
+	if err != nil && !os.IsNotExist(err) {
+		return []Item{}, err
+	}
+
+	results := []Item{}
+
+	for _, nodeDirInfo := range nodeDirs {
+		if !nodeDirInfo.IsDir() {
+			continue
+		}
+		nodeName := filepath.Base(nodeDirInfo.Name())
+		nodeItem := Item{
+			Name: nodeName,
+		}
+		items, err := processDir(p, pdir, filepath.Join(dir, nodeName), processor, selector)
+		nodeItem.Items = items
+		if err != nil {
+			logrus.Warningf("Error processing results entries for node %v, plugin %v: %v", nodeDirInfo.Name(), p.GetName(), err)
+		}
+		results = append(results, nodeItem)
+	}
+
+	return results, nil
+}
+
 func processPluginWithProcessor(p plugin.Interface, baseDir string, processor postProcessor, selector fileSelector) (Item, error) {
 	pdir := path.Join(baseDir, PluginsDir, p.GetName())
 	pResultsDir := path.Join(pdir, ResultsDir)
+	pErrorsDir := path.Join(pdir, ErrorsDir)
 
 	_, isDS := p.(*daemonset.Plugin)
 	results := Item{
 		Name: p.GetName(),
 	}
 
 	if isDS {
-		nodeDirs, err := ioutil.ReadDir(pResultsDir)
+		items, err := processNodesWithProcessor(p, baseDir, pResultsDir, processor, selector)
 		if err != nil {
-			return Item{}, err
+			logrus.Warningf("Error processing results entries for plugin %v: %v", p.GetName(), err)
```

Comment:
TODO: rather than log here, I want to return a slice of errors and just return them all and log at the parent if we want. Awkward to have lower level methods not pass up their errors.

I really just wanted to continue processing after errors so on my first pass I added logging rather than returning.